### PR TITLE
Revive PoI bisection

### PR DESCRIPTION
**Pull request by neysofu**
- Fix failing CI by skipping building `frontend`, which is currently unmaintained.
- Listen on `127.0.0.1` instead of `0.0.0.0`.
- Expand the `Indexer` trait with new methods to fetch metadata from the exposed `graph-node` endpoints. I also refactored much of the `indexer` module and I abstracted away much of the `async_graphql` logic using utility methods. I think the new code looks better, but we should improve the logging situation in `real_indexer.rs` (it's poor and scarce).
- Renamed the `db` module to `store`.
- Added a new event listener to `Store` which uses `sqlx::PgListener` behind the scenes.
- Removed some old dead code that used `eventuals`.
- Rewrote out entire cross-checking and bisection jobs. The new code is a very raw MVP, but at least it kinda works as opposed to the old dead code that wasn't actually called anywhere.

### CODE REVIEW

**Comment by neysofu** [comment]
I just finished splitting the commits into quite atomic changes, reviewing should be easier now.

File: backend/crates/cross-checker/src/main.rs [hunk 1]
```diff
@@ -58,6 +73,61 @@ fn init_tracing() {
     tracing_subscriber::fmt::init();
 }
 
+async fn handle_bisect_requests(
+    store: &store::Store,
+    indexers: watch::Receiver<Vec<Arc<dyn Indexer>>>,
+) -> anyhow::Result<()> {
+    loop {
```
Comment (leoyvens) [inline backend/crates/cross-checker/src/main.rs:80]: Lets remove all the panics in this loop and use a 'log error and ignore' logic.

**Review by leoyvens — APPROVED** [review]
Amazing cleanup work! Just one comment

File: backend/crates/cross-checker/src/main.rs (hunk 1 above)
Comment (neysofu) [reply backend/crates/cross-checker/src/main.rs:80]: Thanks for the reminder, I didn't a proper cleanup of the section after writing the first draft. Doing it now

Comment (neysofu) [reply backend/crates/cross-checker/src/main.rs:88]: Addressed in [27729c35d8fcc410359be7734acb85972c522480](https://github.com/edgeandnode/graphix/pull/70/commits/27729c35d8fcc410359be7734acb85972c522480).

**Pull request merged**